src/netlink.c

```diff
@@ -559,9 +581,9 @@ static int qos_filter_cb (const struct nlmsghdr *nlh, void *args)
       return MNL_CB_ERROR;
     }
 
-    mnl_attr_parse_nested (attr, qos_attr_cb, &bs);
+    mnl_attr_parse_nested (attr, qos_attr_cb, &q_stats);
```

Comment:
Alright. In that case (temporary variable), could you please remove one level of indirection? The pointer-pointer inside the struct is then unnecessary and the struct can be simplified to:

```c
struct qos_stats {
    struct gnet_stats_basic *bs;
    struct gnet_stats_queue *qs;
};
```